base/compiler/typelimits.jl

```diff
@@ -292,13 +294,35 @@ function tmerge(@nospecialize(typea), @nospecialize(typeb))
     suba && subb && return typea
     subb && issimpleenoughtype(typea) && return typea
 
+    # type-lattice for LimitedAccuracy wrapper
+    # the merge create a slightly narrower type than needed, but we can't
+    # represent the precise intersection of causes and don't attempt to
+    # enumerate some of these cases where we could
+    if isa(typea, LimitedAccuracy) && isa(typeb, LimitedAccuracy)
+        if typea.causes ⊆ typeb.causes
+            causes = typeb.causes
+        elseif typeb.causes ⊆ typea.causes
+            causes = typea.causes
+        else
+            causes = union!(copy(typea.causes), typeb.causes)
```

Comment:
Any reason not to use plain `union` here?

Reply:
Yeah, it doesn’t make an IdDict. We should perhaps change that

Reply:
Ah, wait, that is a holdover from when this was a Dict and used `merge`, for union on IdSet, it would be okay.
```
julia> merge(IdDict(), IdDict())
Dict{Any, Any}()
```